src/hardware/serialport/serialport.cpp

```diff
@@ -1018,6 +1018,42 @@ constexpr std::tuple<uint8_t, uint8_t> baud_to_regs(uint32_t baud_rate)
 	return std::make_tuple(transmit_reg, interrupt_reg);
 }
 
+constexpr uint8_t line_control_to_reg(const uint8_t data_bits,
+                                      const char parity_type,
+                                      const uint8_t stop_bits)
+{
+	uint8_t reg = 0;
+
+	switch (data_bits) {
+	case 5: reg |= LCR_DATABITS_5; break;
+	case 6: reg |= LCR_DATABITS_6; break;
+	case 7: reg |= LCR_DATABITS_7; break;
+	default: reg |= LCR_DATABITS_8; break;
+	}
+
+	// Explanation of the parity types:
+	// - Odd: If an odd number of 1s in the data then parity bit is 1
+	// - Even: If an even number of 1s in the data then parity bit is 1
+	// - Mark: the parity bit is expected to always be 1
+	// - Space: the parity bit is expected to always be 0
+	// - None: no parity bit is present or transmitted, which is the default
+	assertm(parity_type >= 'A', "Specify parity type with uppercase");
+	switch (parity_type) {
+	case 'O': reg |= LCR_PARITY_ODD; break;
+	case 'E': reg |= LCR_PARITY_EVEN; break;
+	case 'M': reg |= LCR_PARITY_MARK; break;
+	case 'S': reg |= LCR_PARITY_SPACE; break;
+	default: reg |= LCR_PARITY_NONE; break;
+	}
+
+	if (stop_bits <= 1)
+		reg |= LCR_STOPBITS_1;
+	else
+		reg |= LCR_STOPBITS_MORE_THAN_1;
```

Comment:
Oh, had no idea this is needed. Double checked - LCR_STOPBITS_1 == 0x0, so the behaviour of code is preserved.